src/pluginrc/saltplugins.rc

```diff
@@ -1,17 +1,16 @@
 #!/bin/bash
 
 function check_packages() {
-    RPMS=$1
-    for pkg in $RPMS; do
-	if ! rpm -q $pkg &>/dev/null; then
```

Comment:
I'll keep using `rpm -q $pkg` for checking whether the package exists or not.

Problem is that `rpm -V` might return non-zero retcodes even if the package is installed. i.a. some permission error for a folder according to the package RPM metadata.

We only want to stop the execution of the plugin in case any of the targeted packages are not installed, but not in case they have issues. If they have issues, we want to continue and the issues will be probably shown later once the `validate_rpm_pkg` function is called.

Reply:
Are we really not interested in pkgs that are installed but were modified?

Reply:
@meaksh, I've restored the `rpm -q` usage in the next commit https://github.com/openSUSE/salt-supportconfig/pull/6/commits/4d3b3579c6dd1fcc52b54629bd7ca608d5945c47
I just misunderstood the purpose of `validate_rpm_pkg` function.

Reply:
> Are we really not interested in pkgs that are installed but were modified?

The `rpm -V` checks will still run for all possible Salt packages at https://github.com/openSUSE/salt-supportconfig/blob/39c2acdb84364d7e8f36fd7aab25d638e99fb22b/src/saltgeneral#L18-L20

The `check_packages` function (which only ensures pkg is installed) is executed at the beginning of each report, and it checks the necessary packages are installed in order to perform that report. So is package is missing we need to exit, and if package has changed we also want to run the report anyway.

Reply:
With the latest commit, the functionality is exact as you described. https://github.com/openSUSE/salt-supportconfig/commit/4d3b3579c6dd1fcc52b54629bd7ca608d5945c47

Reply:
I still think it would be slightly better to exit after checking all passed packages and not on the first missing one. Right now it's not a  big deal, we only check one or two packages at a time. If the first one is missing, then installed and supportconfig runs again, it will check the presence of the second one and will be done. If we at some point add more packages to the check (invocations of this function) for whatever reason, we could reduce the number of "report missing -> install one package -> report next missing" cycles. But since we also have the verification check, this might not be in issue after all.

Reply:
So to be clear, I'm okay with the changes you've done.

Reply:
The purpose of this check is to prevent executing the code after this check. For example now it tries to get salt jobs if there is no salt master installed on the host. We just need to ensure if all of listed packages are installed. In all cases check_packages used to check not more than two packages: salt, salt-master. I think there is no reason to walk through all the list. Just exit on the first unavailable package.

Reply:
Okay, then I guess this is fine. Thx!